fix(api): fix float conversion issue when truncating absorbance values.

vegano1 opened the pull request:
# Overview

Converting a float in scientific notation to a string with `str` gives you a string of the scientific notation of the value. We then truncate that string by 5 characters using string formatting, which keeps the one's place, floating point, and 3 decimal places converted back to a float giving us invalid values. Here is a snippet showcasing this behavior

```
>>> value = float(-1.708423405943904e-05)
>>> value
-1.708423405943904e-05
>>> str(value)
'-1.708423405943904e-05'
>>> "{:.5}".format(str(value))
'-1.70'
```

Closes: [EXEC-1295](https://opentrons.atlassian.net/browse/EXEC-1295)

## Test Plan and Hands on Testing

- [x] Make sure we produce correct values when reading absorbance
- [x] Make sure near 0 values expressed in scientific notation are truncated properly

## Changelog

- Don't use string formatting when truncating the absorbance values, use math.floor instead.

## Review requests

- Makes sense?

## Risk assessment
Low

[EXEC-1295]: https://opentrons.atlassian.net/browse/EXEC-1295?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

vegano1 committed:
fix(api): fix float conversion issue when truncating absorbance values.

sfoster1 reviewed (approved):
oh wow that's even worse than i thought, that's insane

vegano1 committed:
dont use format string since this rounds the float
copy the list before calling .reverse() as this modifies the original data
add unit tests for the various cases including near-zero scientific notation floats

CaseyBatten reviewed (approved):
Thanks for finding this, thats crazy! Test case looks good
